Portea el Beamgunexosuit medical beamgun de tg.

Evankhell561 opened the pull request:
**What does this PR do:**
-Portea el Medical Beamgun de tg (solo admin spawn) y le hace unos pequeños cambios para que no sea tan poderoso. Ésta arma curativa ya se encuentra en paradise pero la version de tg agrega particulas para que sea mas vistosa la curacion y tiene un codigo especial para permitir que se pueda utilizar en un mecha. (solo fue porteado para que la version del mecha funione)

-portea el Beamgunexosuit medical beamgun de tg, construible en el exosuit exosuit fabricator con altos requerimientos tecnologicos y de minerales para que no sea facil de obtener. El consumo de energia es super elevado para que no pueda ser abusado ni con una bateria slime y solo se use en casos de emergencia. Quiza es demasiado alto el consumo pero quiero ver como ser comporta esto en nestro servidor primero. No creo que afecte mucho la mecanica de med actual (no podran abusar de el como herramienta curativa principal) pero quiza el consumo es tan grande que lo haga inviable y tampoco aporte nada.

**Images of sprite/map changes (IF APPLICABLE):**
![captura uno](https://user-images.githubusercontent.com/49106866/57589016-9ce68380-74eb-11e9-9863-9d0087c04fc1.png)
![captura2](https://user-images.githubusercontent.com/49106866/57589017-9ce68380-74eb-11e9-97a9-69be6c4c1b07.png)


**Changelog:**
:cl: Evan
add: Portea el Medical Beamgun de tg al juego (solo admin spawn).
add: agrega Beamgunexosuit medical beamgun al juego y al exosuit exosuit fabricator.
/:cl:

Commit:
portea el mechbeam (aun no funciona)

añade el icono, el poder ponerlo en el mech, el disparo, y todo lo demas menos la curacion. (falta el desig y la tecnologia original)

Commit:
corrige un error que sucedio al importar el cofigo de tg

este commit no soluciona nada realmente, solo es un fix del primer commit

Commit:
arregla la curacion

falta aun el desing y el tech origin.

Commit:
agrega el desing. balancea.

agrega el desig, el tech origin y nerfea la curacion.

Commit:
tech origin.

añade el tech origin y el consumo de energia al seleccionar objetivo

Commit:
nerfea el rango del rayo curador.

Commit:
sube un poco el alcance del rayo

sube el alcance del rayo, agrega variables para que el mecha funcione mejor. (en pruebas, tick y tick max)

Commit:
nerfea el rango del rayo

nerfea el rango del rayo.

Commit:
agrega intentos de fixear la energia

Commit:
todo funciona

Commit:
quita comentarios.

Commit:
pule los comentarios de prueba. posible version final.

Commit:
arregla la consistencia tecnologica.

arregla el origin y el req tech. sube el rango de la pistola curativa normal, baja el rango de la del mecha.

Commit:
baja un poco el consumo de bateria

con esto una bateria bluespace se consumiria en diez cargas igualmente pero deja 5kw para que el mecha camine.

Commit:
deshace el commit anterior

regresa el consumo de la baterria a 4k

Commit:
cura más lento

ahora el beamgun cura más lentamente pero cura más de otros daños.

Commit:
Merge branch 'master' into Beamgun

Commit:
Merge branch 'master' into Beamgun

Evankhell561 committed:
Merge branch 'master' into Beamgun

Evankhell561 committed:
Merge branch 'master' into Beamgun

Commit:
el beamgun de tg ahora funciona como en tg. el mounted sigue nerf

El beamgun normal (no mecha) tiene los mismos valores de curacion que en tg.
el beamgun del mecha sigue nerfeado (mitad de la curación)

Commit:
exosuit mounted beamgum tiene los mismos valores del beamgun de paradise

Ajusta los valores de curación del beamgun del mecha para hacerlos iguales al beangun de paradise. dublica el consumo de energia y se cambia el nombre de exosuit beamgun a exosuit prototype beamgun.

Evankhell561 committed:
fix commit anterior

Commit:
ahora el beangum no dispara si no tiene suficiente energia.

Commit:
arregla el nombre enel mechfabricator

ahora el mech fabricator muestra un nombre acorde a los demás equipos médicos

Commit:
mechbeangun MK2

Ahora el mechbeangum ahora gasta energia en el tiempo y no al disparar el rayo. A petición del prefecto cambia los 1 y ceros por booleanos.

iRyzor commented on code/HISPANIA/game/objects/effects/temporary_visuals/miscellaneous.dm:
```diff
@@ -0,0 +1,12 @@
+/obj/effect/temp_visual/healfast //color is white by default, set to whatever is needed
```
Ya existe un efecto que hace esto. 
`/obj/effect/temp_visual/heal`

Evankhell561 replied:
este effecto es una modificación de aquel. El efecto /obj/effect/temp_visual/heal produce bugs visuales con el codigo del beamgun. por eso utilice mi propia modificiacón, intentando no modificar el codigo de  paradise en ningun momento.

iRyzor reviewed (commented):
Revisión rápida #1.

Commit:
elimina el efecto "healfast de hispania, modifica el efecto heal de paradise.

Commit:
cambia el path

cambia el path del mediacalbeangum de tg

Commit:
quita la duración inicial del efecto de heal

Commit:
beamgun mk3

Ahora no cura a silicons, ipc, ni bots.
Ahora puede activarse o desactivarse.
Cambia el nombre en el mech fabricator.
Ahora muestra un mensaje cuando no tiene suficiente energia para funcionar.
